src/compiler/checker.ts

```diff
@@ -9401,7 +9474,19 @@ namespace ts {
 
                 diagnostics.add(createDiagnosticForNodeFromMessageChain(errorNode, errorInfo));
             }
-            return result !== Ternary.False;
+            const answer = result !== Ternary.False;

```

Comment:
If `result` is `Ternary.False` then it is `0`, which is falsy. Below, you can just write `if (headMessage && errorNode && result && source.symbol) {` and leave the last line as `return result !== Ternary.False` to avoid declaring an unnecessary local variable.